### PR TITLE
docs: Document .File variables for pages

### LINKED ISSUE #2154: docs: Document .Page.File
Each page has a `.File` object accessible (of type `source.File`).  We need to document that interface in the Template Variables documentation.

### PR DESCRIPTION
**Pull request by digitalcraftsman**
Fixes #2154.

### CODE REVIEW

**Comment by digitalcraftsman** [comment]
@moorereason I tested the variable names with a theme and the following questions popped up:


- `.File.String`, `.File.Bytes` and `.File.Contents` are accessible but were either nil or empty
- What exactly is the difference between `.File.BaseFileName` and `.File.LogicalName`?
- Again what is the difference between `.File.Section` and `.File.Dir`? The last one just appends a slash à la posts/

**Comment by bep** [comment]
Two items to add here:

1) As @digitalcraftsman  spotted, there are duplicates or almost duplicates here that needs to be fixed. I think @spf13 added some of these over time without much tought.
2) These structs are embeded on Page and, in my head, should be documented with their shortest path. So it is `.UniqueID` (already documented, btw) and not `.File.UniqueID`.

**Comment by moorereason** [comment]
* `Content` is an io.Reader and is probably already at EOF by the time the templates are rendered.  `String` and `Bytes` use `Content`.  I wouldn't document these as template functions.
* Given `foo.txt`, `BaseFilename` is `foo` while `LogicalName` is `foo.txt`.
* I'm assuming `Dir` could return `content/post/dir2/dir3` when `Section` would just return `post`.

I usually look at the test cases to see how these should behave, but there are no tests for much of the `File` package.

**Comment by bep** [comment]
Yes, some of these are really internals, and only exported because they live in a different package.

File: docs/content/templates/variables.md [hunk 1]
```diff
@@ -159,6 +159,17 @@ Also available is `.Site` which has the following:
 **.Site.BuildDrafts** A boolean (Default: false) to indicate whether to build drafts. Defined in the site configuration.<br>
 **.Site.Data**  Custom data, see [Data Files](/extras/datafiles/).<br>
 
+## File Variables
+
+The `.File` variable gives you additional information of pages. Available are the following attributes:
+
+**.File.Path** The original relative path of the page, e.g. content/posts/foo.md<br>
```
Comment (moorereason) [inline docs/content/templates/variables.md]: Use back ticks around the example output for each variable.

File: docs/content/templates/variables.md [hunk 2]
```diff
@@ -159,6 +159,17 @@ Also available is `.Site` which has the following:
 **.Site.BuildDrafts** A boolean (Default: false) to indicate whether to build drafts. Defined in the site configuration.<br>
 **.Site.Data**  Custom data, see [Data Files](/extras/datafiles/).<br>
 
+## File Variables
+
+The `.File` variable gives you additional information of pages. Available are the following attributes:
```
Comment (moorereason) [inline docs/content/templates/variables.md]: We need to make it clear that the `.File` object is only available on Pages and not Nodes (index, lists).

**Comment by digitalcraftsman** [comment]
According to your feedback I updated my pull request.

> @bep: These structs are embeded on Page and, in my head, should be documented with their shortest path. So it is .UniqueID (already documented?) and not .File.UniqueID.

`UniqueID` isn't documented so far. Should I remove the reference here and document it instead under the `.Pages` variables?

**Comment by digitalcraftsman** [comment]
I documented `UniqueID` in 20358f46c8b813ab627cc27e08b1689687e41410 and removed `.File.UniqueID`

**Comment by digitalcraftsman** [comment]
Merged as 04889a5ee6247060cef4bde133a013ce7d4e3b61.

**Pull request closed without merging**